spec/istring.dd

```diff
@@ -0,0 +1,228 @@
+)
+
+$(P Like $(GLINK2 lex, DoubleQuotedString), double-quoted IES literals can
+have escape characters in them. Added to the normal escapes is the ability to
+escape a literal $)
+
+$(P A $ followed by any character other than a left parenthesis is treated as a
+literal $ in the expression, there is no need to escape it.)
+
+$(P The expression inside an $(GLINK InterpolationExpression) is a full D
+expression, and escapes are not needed inside that part of the expression.)
+
+$(H3 $(LNAME2 ies_wysiwyg, Wysiwyg IES Literals))
+
+$(GRAMMAR
+$(GNAME InterpolatedWysiwygLiteral):
+    $(B i$(BACKTICK)) $(GLINK InterpolatedWysiwygCharacters)$(OPT) $(B $(BACKTICK))
+
+$(GNAME InterpolatedWysiwygCharacters):
+    $(GLINK InterpolatedWysiwygCharacter)
+    $(GLINK InterpolatedWysiwygCharacter) $(GSELF InterpolatedWysiwygCharacters)
+
+$(GNAME InterpolatedWysiwygCharacter):
+    $(GLINK2 lex, WysiwygCharacter)
+    $(GLINK InterpolationExpression)
+)
+
+$(P Wysiwyg ("what you see is what you get") IES literals are defined like
+$(GLINK2 lex, WysiwygString) strings, but only support backquote syntax. No
+escapes are recognized inside these literals.)
+
+$(H3 $(LNAME2 ies_token, Token IES Literals))
+
+$(GRAMMAR
+$(GNAME InterpolatedTokenLiteral):
+    $(D iq{) $(GLINK InterpolatedTokenStringTokens)$(OPT) $(D })
+
+$(GNAME InterpolatedTokenStringTokens):
+    $(GLINK InterpolatedTokenStringTokenNoBraces)
+    $(GLINK InterpolatedTokenStringToken) $(GSELF InterpolatedTokenStringTokens)
+
+$(GNAME InterpolatedTokenStringToken):
+    $(GLINK InterpolatedTokenNoBraces)
+    $(B {) $(GSELF InterpolatedTokenStringTokens)$(OPT) $(B })
+
+$(GNAME InterpolatedTokenNoBraces):
+    $(GLINK2 lex, TokenNoBraces)
+    $(GLINK InterpolationExpression)
+)
+
+$(P Like $(GLINK2 lex, TokenString), IES Token literals must contain only
+valid D tokens, with the exception of $(GLINK InterpolationExpression). No
+escapes are recognized.)
+
+$(H2 $(LNAME2 expression_translation, Expression Translation))
+
+$(P When the lexer encounters an Interpolation Expression Sequence, the token
+is translated into a sequence of expressions, which replaces the single token.
+The sequence always starts with the expression
+$(D core.interpolation.InterpolationHeader()) and always ends with
+$(D core.interpolation.InterpolationFooter()))
+
+$(P Each part $(D str) of the token which is literal string data is translated
+into the expression $(D core.interpolation.InterpolatedLiteral!(str)))
+
+$(P Each part $(D \$$(LPAREN)expr$(RPAREN)) of the token which is an $(GLINK
+InterpolationExpression) is translated into the sequence $(D
+core.interpolation.InterpolatedExpression!(expr)$(COMMA) expr).)
+
+------
+// simple version of std.typecons.Tuple
+struct Tuple(T...) { T value; }
+Tuple!T tuple(T...)(T value) { return Tuple!T(value); }
+
+import core.interpolation;
+string name = "John Doe";
+auto items = tuple(i"Hello, $(name), how are you?");
+assert(items == tuple(
```

Comment:
Pretty sure the above lines can be simplified to:
```d
alias AliasSeq(A...) = A;
import core.interpolation;
auto items = i"Hello, $(name), how are you?";
assert(items == AliasSeq!(
```

Reply:
I will try it.

Reply:
Yes, this works, I will simplify.

Reply:
Hm... on second thought, this is doing something that I wasn't intending -- it's creating a compile-time sequence of things, including an alias to `name`. What I really wanted was to show that the sequence is the same at runtime, e.g. what the list of values is. So I think I will leave it as-is.